### ISSUE TITLE
Create an `Interpolate` trait to enable for far more genericness. 

### ISSUE THREAD

**Issue by mitchmindtree** [RustAudio/envelope#1]
`Interpolate` will feature a single method:

```Rust
fn interpolate(&self, other: &Self, scalar: Self::Scalar) -> Self::Output;
```

where the `Self` that is returned is the interpolation of `self` and `other`.

Implement this for all primitives.

**Comment by mitchmindtree** [comment]
Consider using the [interpolation](https://github.com/PistonDevelopers/interpolation/blob/master/src/spatial.rs) crate.

**Comment by bvssvni** [comment]
Interesting. I was thinking of suggesting it, but thought that the need for interpolation here was different.

**Comment by mitchmindtree** [comment]
@bvssvni I like the idea of using Piston's interpolation crate, however I also want to be able to interpolate between all `Int` types too, which I'm not sure you guys would want to support?

The way I want to interpolate integer types is like this:
```Rust
// $T is some Int
// $Scalar is some Float
fn interpolate(&self, other: &$T, scalar: &$Scalar) -> $T {
    *self + ((*other - *self) as $Scalar * scalar).round() as $T
}
```

I realise it's a little rough and that other people may prefer it was implemented in a different way which is why I wasn't sure if you guys would want to support it. The reason I want to support integers in this way is for automation envelopes that have discrete values (in most [DAW](http://en.wikipedia.org/wiki/Digital_audio_workstation)s discrete values are produced by rounding the automation envelope (scalar) to the closest discrete value).

I might consider just doing a `Integer(int)` wrapper struct instead.

**Comment by bvssvni** [comment]
@mitchmindtree I think interpolating integers is fine.

**Comment by bvssvni** [comment]
I opened https://github.com/PistonDevelopers/interpolation/issues/27

**Comment by mitchmindtree** [comment]
@bvssvni awesome, I might put a PR together shortly :+1:

**Comment by mitchmindtree** [comment]
Closed by #2
